git/cmd.py

```diff
@@ -531,6 +533,12 @@ def execute(self, command,
 
         :param with_stdout: If True, default True, we open stdout on the created process
 
+        :param timeout:
```

Comment:
What do you think about renaming the parameter to something even more descriptive, like `kill_after_timeout` to be very clear about the `SIGKILL` nature of the operation.